### PR TITLE
Us 709 voltage level dialog

### PR DESCRIPTION
**Pull request by laurgarn**

### CODE REVIEW

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 1]
```diff
@@ -739,6 +739,16 @@ public void initBinder(WebDataBinder webdataBinder) {
                 .then(studyService.createEquipment(studyUuid, createSubstationAttributes, ModificationType.SUBSTATION_CREATION, nodeUuid)));
     }
 
+    @PutMapping(value = "/studies/{studyUuid}/nodes/{nodeUuid}/network-modification/voltage-levels")
+    @Operation(summary = "create a voltage level in the study network")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The voltage level has been created")})
+    public ResponseEntity<Mono<Void>> createVoltageLevel(@PathVariable("studyUuid") UUID studyUuid,
+        @PathVariable("nodeUuid") UUID nodeUuid,
+        @RequestBody String createSubstationAttributes) {
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/study/server/StudyController.java:747]: createVoltageLevelAttributes

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 2]
```diff
@@ -105,8 +105,8 @@ public void initBinder(WebDataBinder webdataBinder) {
     @PostMapping(value = "/studies/cases/{caseUuid}")
     @Operation(summary = "create a study from an existing case")
     @ApiResponses(value = {
-        @ApiResponse(responseCode = "200", description = "The id of the network imported"),
-        @ApiResponse(responseCode = "409", description = "The study already exist or the case doesn't exists")})
+            @ApiResponse(responseCode = "200", description = "The id of the network imported"),
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/study/server/StudyController.java:108]: keep previous indentations (this code has not been changed ...)
same remarks in following lines

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 3]
```diff
@@ -119,9 +119,9 @@ public void initBinder(WebDataBinder webdataBinder) {
     @PostMapping(value = "/studies", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
     @Operation(summary = "create a study and import the case")
     @ApiResponses(value = {
-        @ApiResponse(responseCode = "200", description = "The id of the network imported"),
-        @ApiResponse(responseCode = "409", description = "The study already exist"),
-        @ApiResponse(responseCode = "500", description = "The storage is down or a file with the same name already exists")})
+            @ApiResponse(responseCode = "200", description = "The id of the network imported"),
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/study/server/StudyController.java:122]: keep previous indentations

File: src/test/java/org/gridsuite/study/server/StudyTest.java [hunk 4]
```diff
@@ -1875,25 +1875,56 @@ public void testCreateSubstation() {
...
 
+    @Test
+    public void testCreateVoltageLevel() {
+        createStudy("userId", CASE_UUID, true);
+        UUID studyNameUserIdUuid = studyRepository.findAll().get(0).getId();
+        UUID rootNodeUuid = getRootNodeUuid(studyNameUserIdUuid);
+        NetworkModificationNode modificationNode = createNetworkModificationNode(studyNameUserIdUuid, rootNodeUuid);
+        UUID modificationNodeUuid = modificationNode.getId();
+
+        // create substation
```
Comment (FranckLecuyer) [inline src/test/java/org/gridsuite/study/server/StudyTest.java:1905]: voltage level

File: src/test/java/org/gridsuite/study/server/StudyTest.java [hunk 5]
```diff
@@ -1875,25 +1875,56 @@ public void testCreateSubstation() {
...
+        String createVoltageLevelAttributes = "{\"voltageLevelId\":\"voltageLevelId1\",\"voltageLevelName\":\"voltageLevelName1\""
+            + ",\"nominalVoltage\":\"379.1\", \"substationId\":\"s1\"}";
+        webTestClient.put()
+            .uri("/v1/studies/{studyUuid}/nodes/{nodeUuid}/network-modification/voltage-levels", studyNameUserIdUuid, rootNodeUuid)
+            .bodyValue(createVoltageLevelAttributes)
+            .exchange()
+            .expectStatus().isOk();
+        checkEquipmentCreationMessagesReceived(studyNameUserIdUuid, rootNodeUuid, new HashSet<>());
+
+        // create substation on modification node child of root node
```
Comment (FranckLecuyer) [inline src/test/java/org/gridsuite/study/server/StudyTest.java:1915]: voltage level

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/study/server/StudyController.java (hunk 1 above)
Comment (laurgarn) [reply src/main/java/org/gridsuite/study/server/StudyController.java]: done

File: src/main/java/org/gridsuite/study/server/StudyController.java (hunk 3 above)
Comment (laurgarn) [reply src/main/java/org/gridsuite/study/server/StudyController.java]: done

File: src/test/java/org/gridsuite/study/server/StudyTest.java (hunk 4 above)
Comment (laurgarn) [reply src/test/java/org/gridsuite/study/server/StudyTest.java]: done

File: src/test/java/org/gridsuite/study/server/StudyTest.java (hunk 5 above)
Comment (laurgarn) [reply src/test/java/org/gridsuite/study/server/StudyTest.java]: done

**Review by FranckLecuyer — APPROVED** [review]

**Review by JacquesBorsenberger — APPROVED** [review]

**Pull request merged**